packages/@angular/cli/ember-cli/lib/cli/cli.js

```diff
@@ -76,169 +83,61 @@ class CLI {
    * @return {Promise}
    */
   run(environment) {
-    let shutdownOnExit = null;
-
-    return Promise.resolve().then(() => {
-      let args = environment.cliArgs.slice();
-
-      if (args.length === 0) {
-        args[0] = 'help';
-      }
-
-      if (args[0] === '--help') {
-        if (args.length === 1) {
-          args[0] = 'help';
-        } else {
-          args.shift();
-          args.push('--help');
-        }
-      }
-
-      let commandName = args.shift();
-      let commandArgs = args;
-      let helpOptions;
-
-      let CurrentCommand = lookupCommand(environment.commands, commandName, commandArgs, {
-        project: environment.project,
-        ui: this.ui,
-      });
-
-      let command = new CurrentCommand({
-        ui: this.ui,
-        analytics: this.analytics,
-        commands: environment.commands,
-        tasks: environment.tasks,
-        project: environment.project,
-        settings: environment.settings,
-        testing: this.testing,
-        cli: this,
-      });
-
-      getOptionArgs('--verbose', commandArgs).forEach(arg => {
-        process.env[`EMBER_VERBOSE_${arg.toUpperCase()}`] = 'true';
-      });
 
-      if (!this.testing) {
-        let skipInstallationCheck = commandArgs.indexOf('--skip-installation-check') !== -1;
-        if (environment.project.isEmberCLIProject() && !skipInstallationCheck) {
-          const InstallationChecker = require('../models/installation-checker');
-          new InstallationChecker({ project: environment.project }).checkInstallations();
+    const logger = new logging.IndentLogger('cling');
+    let loggingSubscription;
+    if (!this.testing) {
+      loggingSubscription = this.initializeLogging(logger);
+    }
+    const context = {
+      project: environment.project,
+      ui: this.ui,
+    };
+    return runCommand(environment.commands, environment.cliArgs, logger, context)
```

Comment:
Are we doing `await`/`async` in this PR?

Reply:
Yes, I can refactor, but I wanted it working first.